packages/blocks/src/blocks.ts

```diff
@@ -1,19 +1,125 @@
-import type {
-  Block,
-  Blocks,
-  BlocksOptions,
-  Cell,
-  InferAllButLastArrayElements,
-  InferLastArrayElement,
-  InvalidCellMessage,
-} from '@/types';
 import { dump, log, raise } from '@ballerine/common';
+import { ComponentProps, FunctionComponent, ReactNode } from 'react';
+
+/**
+ * A cell may have any property as long as it has a `type` property to discriminate a union by.
+ */
+export type Cell = { type: string } & {
+  [key: string]: unknown;
+};
+
+export type Block = Array<Cell>;
+
+export type Blocks = Array<Block>;
+
+/**
+ * Allow the consumer of `@ballerine/blocks` to register their own cell types.
+ * @example
+ * declare module '@ballerine/blocks' {
+ *    interface BlocksClient {
+ *      cells: typeof blocks; // createBlocks<TCell>();
+ *    }
+ *  }
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface BlocksClient {
+  // cells: ReturnType<typeof createBlocks<TCell>>;
+}
 
-export const createBlocks = <TCell extends Cell | InvalidCellMessage = InvalidCellMessage>(
-  options: BlocksOptions = {},
-) =>
-  new BlocksBuilder<// @ts-expect-error - A generic should always be provided.
-  TCell>([], options);
+/**
+ * Infer the generic passed into an instance of `BlocksBuilder`.
+ */
+type InferBlocksBuilderGeneric<TClass extends BlocksBuilder<any>> = TClass extends BlocksBuilder<
+  infer TCell
+>
+  ? TCell
+  : never;
+
+/**
+ * Infer the generic passed into the instance of `BlocksBuilder` registered by the consumer of `@ballerine/blocks`.
+ */
+export type Cells =
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  InferBlocksBuilderGeneric<BlocksClient['cells']>;
```

Comment:
Fix 2/2. `BlocksClient` was previously `Blocks`, `Blocks` does not have the `cells` property registered by the package consumer.